.github/scripts/comment_on_pr.py

```diff
@@ -104,14 +121,16 @@ def check_code_changes(before_code, after_code):
         print(f"Error checking code changes for file {filename}: {e}")
```

Comment:
The code changes seem to be aimed at improving the handling of GitHub pull request comments and integrating a code change checking mechanism. However, there are a few areas that need attention:

1. The `continue` statement in the `check_code_changes` function is not properly indented, which will result in a `SyntaxError`.
2. The `response` variable in the `check_code_changes` function is not defined within the function scope.
3. The `comment_body` variable is set to `response` in the second code snippet, but it's not clear what `response` refers to at that point. It should be set to the actual comment body obtained from the response.
4. The `comment_data` dictionary in the second code snippet uses `commit_id` and `line` instead of `position`. Ensure that this change is consistent with the API requirements.
5. The `side` parameter is added to the `comment_data` dictionary, which might be specific to the API you are using. Make sure this parameter is supported by the GitHub API.

Here's a revised version of the code with some of these issues addressed:

```python
# before:
print(f"Response content: {response.text}")
raise

# 获取最新的 commit_id
pr_url = f"{api_url}/pulls/{pr_number}"
pr_response = requests.get(pr_url, headers=headers)

if pr_response.status_code != 200:
    print(f"Error fetching PR details: {pr_response.status_code} {pr_response.text}")
    pr_response.raise_for_status()

try:
    pr_details = pr_response.json()
    commit_id = pr_details['head']['sha']
except json.JSONDecodeError as e:
    print(f"Error decoding JSON: {e}")
    print(f"Response content: {pr_response.text}")
    raise

# 利用codegeex
api_key = "da1f20fa26ff33bf88deec61ff67c743.hpZFBH5QWUev0kXZ"

def check_code_changes(before_code, after_code):
    try:
        # Assuming 'response' is obtained from some API call here
        response = some_api_call()
        comment_body = response.get('choices', [{}])[0].get('message', {}).get('content', '这里的更改需要注意。')
    except Exception as e:
        print(f"Error checking code changes for file {filename}: {e}")
        return  # Fixed the indentation and replaced 'continue' with 'return'

    # 添加评论
    comment_url = f"{api_url}/pulls/{pr_number}/comments"
    comment_data = {
        "body": comment_body,
        "path": filename,
        "commit_id": commit_id,
        "line": new_start_line,  # 使用 line 而不是 position
        "side": "RIGHT"
    }

    comment_response = requests.post(comment_url, json=comment_data, headers=headers)
    # Handle comment_response as needed
```

Make sure to replace `some_api_call()` with the actual API call that provides the `response` object. Also, ensure that the `commit_id`, `line`, and `side` parameters are correctly understood and supported by the GitHub API you are using.